android/app/src/main/java/com/zulipmobile/ShareContent.java

```diff
@@ -0,0 +1,93 @@
+package com.zulipmobile;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.core.content.FileProvider;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+
+import org.apache.commons.io.IOUtils;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Allows Sending Content from zulip to other apps.
+ */
+
+public class ShareContent extends ReactContextBaseJavaModule {
+    private String TAG = "ShareContent";
+
+    public ShareContent(ReactApplicationContext reactContext) {
+        super(reactContext);
+    }
+
+    @Override
+    public String getName() {
+        return "ShareContent";
+    }
+
+    @ReactMethod
+    public void shareContent(@Nullable ReadableMap source) {
+        try {
+            String urlString = source.getString("uri");
+            String[] urlSplit = urlString.split("/");
+            String fileName = urlSplit[urlSplit.length - 1];
+            URL url = new URL(urlString);
+            String auth = source.getMap("headers").getString("Authorization");
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+            conn.addRequestProperty("Authorization", auth);
+            String mimeType = conn.getContentType();
```

Comment:
Do we want to limit the possible mimetypes we'll share? Possibly it makes sense to just share whatever we get, I don't know if there are security implications there or anything.

I think probably it makes sense to leave this as-is, I just wanted to bring it up in case anyone knows more about this than I do and has a reason to restrict mimetypes.